src/test/java/org/dcsa/ebl/service/impl/ShippingInstructionServiceImplTest.java

```diff
@@ -0,0 +1,397 @@
+                assertEquals(
+                    shippingInstruction.getDocumentStatus(),
+                    argumentCaptor.getValue().getDocumentStatus());
+              })
+          .verifyComplete();
+    }
+
+    @Test
+    @DisplayName("Method should save a shallow shipping instruction and return shipping response")
+    void testCreateShippingInstructionShallow() {
+
+      shippingInstructionTO.setCarrierBookingReference(null);
+      shippingInstructionTO.setPlaceOfIssue(null);
+      shippingInstructionTO.setDocumentParties(null);
+      shippingInstructionTO.setReferences(null);
+      shippingInstructionTO.setShipmentEquipments(null);
+
+      when(shippingInstructionRepository.save(any())).thenReturn(Mono.just(shippingInstruction));
+      when(shipmentEventService.create(any()))
+          .thenAnswer(arguments -> Mono.just(arguments.getArguments()[0]));
+
+      ArgumentCaptor<ShippingInstruction> argumentCaptor =
+          ArgumentCaptor.forClass(ShippingInstruction.class);
+
+      StepVerifier.create(
+              shippingInstructionServiceImpl.createShippingInstruction(shippingInstructionTO))
+          .assertNext(
+              b -> {
+                assertEquals(
+                    shippingInstruction.getShippingInstructionID(), b.getShippingInstructionID());
+                assertEquals("Received", b.getDocumentStatus().getValue());
+                assertNotNull(b.getShippingInstructionCreatedDateTime());
+                assertNotNull(b.getShippingInstructionUpdatedDateTime());
+
+                verify(shippingInstructionMapper)
+                    .shippingInstructionToShippingInstructionResponseTO(argumentCaptor.capture());
+                assertEquals(
+                    shippingInstruction.getShippingInstructionID(),
+                    argumentCaptor.getValue().getShippingInstructionID());
+                assertEquals(
+                    shippingInstruction.getPlaceOfIssueID(),
+                    argumentCaptor.getValue().getPlaceOfIssueID());
+                assertEquals(
+                    shippingInstruction.getDocumentStatus(),
+                    argumentCaptor.getValue().getDocumentStatus());
+
+                verify(locationService, never()).createLocationByTO(any(), any());
+                verify(shipmentRepository, never()).findByCarrierBookingReference(any());
+                verify(ShipmentEquipmentService, never())
+                    .createShipmentEquipment(any(), any(), any());
+                verify(documentPartyService, never())
+                    .createDocumentPartiesByShippingInstructionID(any(), any());
+                verify(referenceService, never())
+                    .createReferencesByShippingInstructionIDAndTOs(any(), any());
+              })
+          .verifyComplete();
+    }
+
+    @Test
+    @DisplayName("Failing creation a shipment event should result in an error")
+    void testShipmentEventFailedShouldResultInError() {
```

Comment:
Is this testing both POST and PUT? In both cases a ShipmentEvent should be created

Reply:
This is testing the implementation, not the endpoint.

Reply:
I know ;-) - what I meant was - does it both test update and create - or is it done using the same method above?